src/http.zig

```diff
@@ -3358,6 +3396,94 @@ noinline fn sendInitialRequestPayload(this: *HTTPClient, comptime is_first_call:
     };
 }
 
+pub fn flushStream(this: *HTTPClient, comptime is_ssl: bool, socket: NewHTTPContext(is_ssl).HTTPSocket, data: []const u8) void {
```

Comment:
Can we simplify this function to be clearer and have less duplicate code? I'm worried there are bugs here caused by confusing control flow.